README.md

```diff
@@ -72,19 +69,117 @@ require('calendar')
 ```
 app/javascript/calendar/index.js // gets loaded by require('calendar')
 app/javascript/calendar/components/grid.jsx
+app/javascript/calendar/styles/grid.sass
 app/javascript/calendar/models/month.js
 ```
 
-But it could also look a million other ways. The only convention that Webpacker enforces is the
-one where entry points are automatically configured by the files in `app/javascript/packs`.
+```erb
+<%# app/views/layout/application.html.erb %>
+<%= javascript_pack_tag 'calendar' %>
+<%= stylesheet_pack_tag 'calendar' %>
+```
+
+But it could also look a million other ways.
+
+## Advanced Configuration
+
+By default, webpacker offers simple conventions for where the webpack configs, javascript app files and compiled webpack bundles will go in your rails app,
+but all these options are configurable from `config/webpacker.yml`
+that comes with webpacker.
+
+```yml
+default: &default
+  paths:
+    src_path: app/javascript
+    config_path: config/webpack
+    node_modules_path: node_modules
+    dist_dir: packs
+    dist_path: public/packs
+
+  dev_server:
+    enabled: true
+    host: localhost
+    port: 8080
+    compress: true
+
+production:
+  <<: *default
+
+development:
+  <<: *default
+
+test:
+  <<: *default
+```
 
+**For ex:** if you rename `packs` directory inside `app/javascript` from `packs` to `bundles`, make sure you also update your `dist_dir` and `dist_path`.
+
+```yml
+default: &default
+  paths:
+    dist_dir: bundles
+    dist_path: public/bundles
+```
+
+## Linking to static assets
+
+Static assets like images, fonts and stylesheets support is enabled out-of-box so, you can link them into your javascript app code and have them compiled automatically.
+
+```js
+// React component example
+// app/javascripts/packs/hello_react.jsx
+import React from 'react'
+import ReactDOM from 'react-dom'
+import helloIcon from '../hello_react/images/icon.png'
+import './hello-react.sass'
+
+const Hello = props => (
+  <div className="hello-react">
+    <img src={helloIcon} alt="hello-icon" />
+    <p>Hello {props.name}!</p>
+  </div>
+)
+```
+
+under the hood webpack uses [extract-text-webpack-plugin](https://github.com/webpack-contrib/extract-text-webpack-plugin) plugin to extract all the referenced styles and compile it into a separate `[pack_name].css` bundle so that within your view you can use the `stylesheet_pack_tag` helper,
+
+```erb
+<%= stylesheet_pack_tag 'hello_react' %>
+```
+
+## Getting asset path
+
+Webpacker provides `asset_pack_path` helper to get the path of any given asset that's been compiled by webpack. For ex, if you want to create a `<link rel="prefetch">` or `<img />`
+for an asset used in your pack code you can reference them like this in your view,
+
+```erb
+<%= asset_pack_path 'hello_react.css' %>
+<% # => "/packs/hello_react.css" %>
+<%= asset_pack_path 'hello_react', type: :stylesheet %>
+<% # => "/packs/hello_react.css" %>
+<%= asset_pack_path 'hello_react', type: :javascript %>
```

Comment:
Gotcha. We don't need to advertise that. It's kinda particular to how asset pipeline does things.